zebra-consensus/src/block.rs

```diff
@@ -271,10 +272,10 @@ where
             match state_service
                 .ready()
                 .await
-                .map_err(VerifyBlockError::Commit)?
+                .expect("poll_ready method of state service and Ready::<StateService>::poll will never return Err")
```

Comment:
For your information - no changes needed:

This change locks us into always having the state service ready.

That might be ok, because our design requires the service is either ready: `Poll::Ready(Ok(_))`, or permanently failed: `Poll::Ready(Err(_))`. The state service can never be temporarily paused: `Poll::Pending`.

The state needs to commit blocks in order. So we want to pause higher blocks, but still let lower blocks commit. But Tower services can only pause all requests,  so we can't use `poll_ready()` to do that.

So as long as we're sure we'll never error here, even on shutdown, it would be ok to do this.

Reply:
Got it! I was very uncertain about this part, so this is good to know.

Reply:
I am also unsure about this!